File: packages/react-native-vision-camera/ios/Frame Processors/HybridNativeThread.swift
///
/// HybridNativeThread.swift
/// VisionCamera
/// Copyright © 2025 Marc Rousavy @ Margelo
///

import Foundation
import NitroModules

class HybridNativeThread: HybridNativeThreadSpec {
  let queue: DispatchQueue

  init(queue: DispatchQueue) {
    self.queue = queue
    super.init()
  }
  
  var id: String {
    return queue.label
  }
  
  func runOnThread(task: @escaping () -> Void) throws {
    queue.async(execute: task)
  }
}

File: packages/react-native-vision-camera/android/src/main/java/com/margelo/nitro/camera/hybrids/HybridNativeThread.kt
package com.margelo.nitro.camera.hybrids

import com.margelo.nitro.camera.HybridNativeThreadSpec
import java.util.concurrent.Executor

class HybridNativeThread(
  val executor: Executor,
) : HybridNativeThreadSpec() {
  override val id: String
    get() = executor.toString()

  override fun runOnThread(task: () -> Unit) {
    executor.execute(task)
  }
}
